drivers/dma/dma_rpi_pico.c

```diff
@@ -0,0 +1,384 @@
+					 !!(dma_cfg->channel_priority));
+
+	data->channels[channel].callback = dma_cfg->dma_callback;
+	data->channels[channel].user_data = dma_cfg->user_data;
+	data->channels[channel].direction = dma_cfg->channel_direction;
+
+	return 0;
+}
+
+static int dma_rpi_pico_reload(const struct device *dev, uint32_t ch, uint32_t src, uint32_t dst,
+			       size_t size)
+{
+	const struct dma_rpi_pico_config *cfg = dev->config;
+	struct dma_rpi_pico_data *data = dev->data;
+
+	if (ch >= cfg->channels) {
+		LOG_ERR("reload channel must be < %" PRIu32 " (%" PRIu32 ")", cfg->channels, ch);
+		return -EINVAL;
+	}
+
+	if (dma_channel_is_busy(ch)) {
+		return -EBUSY;
+	}
+
+	data->channels[ch].source_address = (void *)src;
+	data->channels[ch].dest_address = (void *)dst;
+	data->channels[ch].block_size = size;
+	dma_channel_configure(ch, &data->channels[ch].config, data->channels[ch].dest_address,
+			      data->channels[ch].source_address, data->channels[ch].block_size,
+			      true);
+
+	return 0;
+}
+
+static int dma_rpi_pico_start(const struct device *dev, uint32_t ch)
+{
+	const struct dma_rpi_pico_config *cfg = dev->config;
+	struct dma_rpi_pico_data *data = dev->data;
+
+	if (ch >= cfg->channels) {
+		LOG_DBG("start channel must be < %" PRIu32 " (%" PRIu32 ")", cfg->channels, ch);
+		return -EINVAL;
+	}
+
+	dma_irqn_acknowledge_channel(dma_rpi_pico_channel_irq(dev, ch), ch);
+	dma_irqn_set_channel_enabled(dma_rpi_pico_channel_irq(dev, ch), ch, true);
+
+	dma_channel_configure(ch, &data->channels[ch].config, data->channels[ch].dest_address,
+			      data->channels[ch].source_address, data->channels[ch].block_size,
+			      true);
+
+	return 0;
+}
+
+static int dma_rpi_pico_stop(const struct device *dev, uint32_t ch)
+{
+	const struct dma_rpi_pico_config *cfg = dev->config;
+
+	if (ch >= cfg->channels) {
+		LOG_DBG("stop channel must be < %" PRIu32 " (%" PRIu32 ")", cfg->channels, ch);
+		return -EINVAL;
+	}
+
+	dma_irqn_set_channel_enabled(dma_rpi_pico_channel_irq(dev, ch), ch, false);
+	rpi_pico_dma_channel_clear_error_flags(dev, ch);
+
+	/*
+	 * Considering the possibility of being called in an interrupt context,
+	 * it does not wait until the abort bit becomes clear.
+	 * Ensure the busy status is canceled with dma_get_status
+	 * before the next transfer starts.
+	 */
+	rpi_pico_dma_channel_abort(dev, ch);
+
+	return 0;
+}
+
+static int dma_rpi_pico_get_status(const struct device *dev, uint32_t ch, struct dma_status *stat)
+{
+	const struct dma_rpi_pico_config *cfg = dev->config;
+	struct dma_rpi_pico_data *data = dev->data;
+
+	if (ch >= cfg->channels) {
+		LOG_DBG("channel must be < %" PRIu32 " (%" PRIu32 ")", cfg->channels, ch);
+		return -EINVAL;
+	}
+
+	stat->pending_length = 0;
+	stat->dir = data->channels[ch].direction;
+	stat->busy = dma_channel_is_busy(ch);
+
+	return 0;
+}
+
+static bool dma_rpi_pico_api_chan_filter(const struct device *dev, int ch, void *filter_param)
+{
+	uint32_t filter;
+
+	if (!filter_param) {
```

Comment:
If this is a possible runtime error condition, LOG_ERR.

If this is a programming error __ASSERT()

Reply:
The function caller can give the NULL as filter_pattern args.
It is need to handle as the runtime error here.